specs/altair/beacon-chain.md

```diff
@@ -0,0 +1,699 @@
+# Ethereum 2.0 Altair
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Custom types](#custom-types)
+- [Constants](#constants)
+  - [Participation flag indices](#participation-flag-indices)
+  - [Participation flag fractions](#participation-flag-fractions)
+  - [Misc](#misc)
+- [Configuration](#configuration)
+  - [Updated penalty values](#updated-penalty-values)
+  - [Misc](#misc-1)
+  - [Time parameters](#time-parameters)
+  - [Domain types](#domain-types)
+- [Containers](#containers)
+  - [Modified containers](#modified-containers)
+    - [`BeaconBlockBody`](#beaconblockbody)
+    - [`BeaconState`](#beaconstate)
+  - [New containers](#new-containers)
+    - [`SyncCommittee`](#synccommittee)
+- [Helper functions](#helper-functions)
+  - [`Predicates`](#predicates)
+    - [`eth2_fast_aggregate_verify`](#eth2_fast_aggregate_verify)
+  - [Misc](#misc-2)
+    - [`get_flag_indices_and_numerators`](#get_flag_indices_and_numerators)
+    - [`add_flag`](#add_flag)
+    - [`has_flag`](#has_flag)
+  - [Beacon state accessors](#beacon-state-accessors)
+    - [`get_sync_committee_indices`](#get_sync_committee_indices)
+    - [`get_sync_committee`](#get_sync_committee)
+    - [`get_base_reward`](#get_base_reward)
+    - [`get_unslashed_participating_indices`](#get_unslashed_participating_indices)
+    - [`get_flag_deltas`](#get_flag_deltas)
+    - [New `get_inactivity_penalty_deltas`](#new-get_inactivity_penalty_deltas)
+  - [Beacon state mutators](#beacon-state-mutators)
+    - [New `slash_validator`](#new-slash_validator)
+  - [Block processing](#block-processing)
+    - [Modified `process_attestation`](#modified-process_attestation)
+    - [Modified `process_deposit`](#modified-process_deposit)
+    - [Sync committee processing](#sync-committee-processing)
+  - [Epoch processing](#epoch-processing)
+    - [Justification and finalization](#justification-and-finalization)
+    - [Leak scores](#leak-scores)
+    - [Rewards and penalties](#rewards-and-penalties)
+    - [Slashings](#slashings)
+    - [Participation flags updates](#participation-flags-updates)
+    - [Sync committee updates](#sync-committee-updates)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+Altair is a patch implementing the first hard fork to the beacon chain.
+It has four main features:
+
+* Light client support via sync committees
+* Incentive accounting reforms, reducing spec complexity
+  and [TODO] reducing the cost of processing chains that have very little or zero participation for a long span of epochs
+* Update penalty configuration values, moving them toward their planned maximally punitive configuration
+* Fork choice rule changes to address weaknesses recently discovered in the existing fork choice
```

Comment:
Is this referring to the block-slot fix? If so, should remove also remove this line?